src/lib.rs

```diff
@@ -373,6 +395,83 @@ pub fn return_fdtable_copy(cageid: u64) -> HashMap<u64, FDTableEntry> {
     fdtable.get(&cageid).unwrap().clone()
 }
 
+/* SELECT()
+*   fd_set is used in the Linux select system call to specify the file descriptor 
+*   to be monitored. fd_set is actually a bit array, each bit of which represents 
+*   a file descriptor. We use Vec to express the fd_set of the virtual file descriptor 
+*   in Lind, and expand the conversion function between lind fd_set and kernel fd_set.
+*
+*   We won't change value inside of Vec, so only using borrow immutable reference
+*   for virtual_to_real_set
+*   Reference: https://github.com/rust-lang/libc/issues/475
+*/
+pub fn virtual_to_real_set(cageid: u64, virtualfds: &Vec<u64>) -> fd_set {
+    let fdtable = GLOBALFDTABLE.lock().unwrap();
+    if !fdtable.contains_key(&cageid) {
+        panic!("Unknown cageid in fdtable access");
+    }
+
+    let mut real_set = unsafe { std::mem::zeroed::<fd_set>() };
+
+    for &virtualfd in virtualfds {
+        let real_fd = translate_virtual_fd(cageid, virtualfd).unwrap() as RawFd;
+        unsafe { FD_SET(real_fd, &mut real_set) };
+    }
+
+    real_set    
+}
+
+/*
+*   Implement the converse one incase we need this in future 
+*/
+pub fn real_to_virtual_set(cageid: u64, real_fds: &fd_set) -> Vec<u64> {
+    let fdtable = GLOBALFDTABLE.lock().unwrap();
+    if !fdtable.contains_key(&cageid) {
+        panic!("Unknown cageid in fdtable access");
+    }
+
+    let mut virtual_set = Vec::new();
+
+    if let Some(virtual_fds) = fdtable.get(&cageid) {
+        for (&virtual_fd, entry) in virtual_fds {
+            let real_fd = entry.realfd as RawFd;
+            if unsafe { FD_ISSET(real_fd, real_fds) } {
+                virtual_set.push(virtual_fd);
+            }
+        }
+    }
+
+    virtual_set
+}
+
+/* POLL()
+*   In Linux, there is a specific structure pollfd used to pass file descriptors and their 
+*   related event information. Through the poll() function, multiple file descriptors can be 
+*   monitored at the same time, and different event monitoring can be set for each file 
+*   descriptor. We implement our PollStruct and related helper functions to do translation 
+*   between virtual fd and kernel fd, in order to use kernel system calls.
+*/
+pub struct PollStruct {
+    pub virtual_fd: u64,
+    pub events: i16,
+    pub revents: i16,
+}
+
+/* [TODO]
+*   PollStruct has multiple..? 
```

Comment:
Likely also needs to take an array...

Reply:
Do you think it makes sense to handle a vector of `PollStruct` in the `poll_syscall`?

Reply:
Yes, I would add a vector here as well